itsnameissteven/metAPI
# Feature/update category selection

## What does this PR do?

- Changes category selection from a long dropdown menu to a more user friendly tag system which allows users to filter the apis by selecting/deselecting one or more tags at the top of the page.
- Refactors change handler methods in the FilterForm component and breaks up the filter method in the App component for improved readability and closer adherence to SRP.

## How should it be tested in the UI?

- Conduct searches by changing form elements in different orders to verify that search results are displaying as expected. Try searching for all the Anime and Government apis with an Auth value of OAuth for example.
